src/middlewared/middlewared/etc_files/loader.py

```diff
@@ -125,5 +127,19 @@ def generate_ec2_config(middleware):
         ]
 
 
+def generate_dual_nvdimm_config(middleware):
+    data = middleware.call_sync('system.dmidecode_info')['system-product-name']
+
+    product = data['system-product-name']
+    current_vers = version.parse(data['system-version'])
```

Comment:
I dont know what all our supported platforms return in this field.

Might be wise to catch an exception here and return early